content/post/flinksql-time.adoc

```diff
@@ -0,0 +1,1033 @@
+We do have a time attribute on the table—`flink_proc_time`
+
+[source,sql]
+----
+Flink SQL> DESCRIBE orders_kafka;
++------------------+-----------------------------+-------+---------------+---------------------------+-----------+
+|             name |                        type |  null |           key |                    extras | watermark |
++------------------+-----------------------------+-------+---------------+---------------------------+-----------+
+|         order_id |                         INT | FALSE | PRI(order_id) |                           |           |
+|      customer_id |                         INT |  TRUE |               |                           |           |
+|     total_amount |              DECIMAL(10, 2) |  TRUE |               |                           |           |
+|           status |                      STRING |  TRUE |               |                           |           |
+|       created_at |                TIMESTAMP(3) |  TRUE |               |                           |           |
+|       shipped_at |                TIMESTAMP(3) |  TRUE |               |                           |           |
+| shipping_address |                      STRING |  TRUE |               |                           |           |
+|   payment_method |                      STRING |  TRUE |               |                           |           |
+|      record_time |            TIMESTAMP_LTZ(3) |  TRUE |               | METADATA FROM 'timestamp' |           |
+|  flink_proc_time | TIMESTAMP_LTZ(3) *PROCTIME* | FALSE |               |           AS `PROCTIME`() |           |
++------------------+-----------------------------+-------+---------------+---------------------------+-----------+
+10 rows in set
+----
+
+So let's use that in the query and see what happens:
+
+[source,sql]
+----
+SELECT  window_start,
+        window_end,
+        COUNT(*) as event_count
+FROM TABLE(
+        TUMBLE(TABLE orders_kafka,
+                DESCRIPTOR(flink_proc_time),
+                INTERVAL '1' MINUTE)
+        )
+GROUP BY window_start, window_end;
+----
+
+At first, we get nothing:
+
+[source,sql]
+----
++----+-------------------------+-------------------------+----------------------+
+| op |            window_start |              window_end |          event_count |
++----+-------------------------+-------------------------+----------------------+
+
+----
+
+That's because Flink waits for the window to close before issuing the result:
+
+[source,sql]
+----
++----+-------------------------+-------------------------+----------------------+
+| op |            window_start |              window_end |          event_count |
++----+-------------------------+-------------------------+----------------------+
+| +I | 2025-04-25 15:11:00.000 | 2025-04-25 15:12:00.000 |                    2 |
+
+----
+
+Let's look closely at the window timestamp though.
+The question we've answered here is _how many records arrived for processing per minute_.
```

Comment:
I think the question we've answered is _how many records were processed each minute_. To my way of thinking, a window based on `record_time` would tell us how many records arrived each minute.

Reply:
good point. fixed.